apps/ejabberd/test/aws_signature_v4_SUITE.erl

```diff
@@ -0,0 +1,130 @@
+-module(aws_signature_v4_SUITE).
+-compile([export_all]).
+
+-include_lib("common_test/include/ct.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+-define(ACCESS_KEY_ID, "AKIAIAOAONIULXQGMOUA").
+-define(SECRET_ACCESS_KEY, "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").
+-define(REGION, "eu-west-1").
+-define(HOST, "s3-eu-west-1.amazonaws.com").
+-define(DATE, {{2014, 1, 2}, {0, 0, 0}}).
+-define(BUCKET, "testbucket").
+-define(EXPIRATION_TIME, 3600).
+
+all() -> [
+          signs_basic_url,
+          signs_urls_with_additional_headers,
+          handles_unicode_path,
+          formats_datetime_timestamp,
+          pads_datetime_timestamp,
+          formats_date_timestamp,
+          pads_date_timestamp,
+          composes_scope_from_values,
+          does_not_encode_unreserved_set,
+          does_encode_reserved_set,
+          does_encode_utf8_characters
+         ].
+
+%% Tests
+
+%% sign
+
+signs_basic_url(Config) ->
+    Path = <<"/", ?BUCKET, "/test/key">>,
+    Signature = sign(Path, ?config(queries, Config), ?config(headers, Config)),
+    ?assertEqual(<<"85bf825e9b67d7838f354b74d97253317584919d8a77b7b50675c2005f003f1f">>,
+                 Signature).
+
+signs_urls_with_additional_headers(Config) ->
+    Headers = maps:put(<<"content-length">>, <<"12">>, ?config(headers, Config)),
+    Queries = queries(Headers, ?DATE, ?EXPIRATION_TIME),
+    Path = <<"/", ?BUCKET, "/test/key">>,
+    Signature = sign(Path, Queries, Headers),
+    ?assertEqual(<<"0ac4c8569e7a9e50af1a34db59678b4f23b98ff9734719b2727fe2e971626d72">>,
+                 Signature).
+
+handles_unicode_path(Config) ->
+    Datetime = {{2017, 1, 3}, {14, 47, 32}},
+    Queries = queries(headers(), Datetime, 900),
+    Path = <<"/♠♣♥♦/test/key"/utf8>>,
```

Comment:
I think it may be a bug fixed in newer versions of Elvis; locally on Elvis ref 2c92119 I get no warnings when rocking.